.github/workflows/trigger-profiling.yaml

```diff
@@ -0,0 +1,154 @@
+        run-number: ${{ github.run_number }}
+        sha: ${{ steps.sha.outputs.result }}
+      secrets:
+        ## The token provided needs contents and pages access to the target repo
+        ## Token can be (re)generated by a member of the UCL organisation, 
+        ## the current member is the rc-softdev-admin.
+        ## [10-07-2023] The current token will expire 10-07-2024
+        profiling-access-token: ${{ secrets.PROFILING_REPO_ACCESS }}
+    - name: Get comment-bot token
+      if: always() && steps.has_permissions.outputs.result == 'true'
+      id: get_comment_bot_token
+      uses: peter-murray/workflow-application-token-action@e8782d687a306fb13d733244d0f2a50e272d3752
+      with:
+        application_id: ${{ secrets.COMMENT_BOT_APP_ID }}
+        application_private_key: ${{ secrets.COMMENT_BOT_APP_PRIVATE_KEY }}
+        organization: ${{ inputs.application-organization }}
+    - name: Create comment with test result and link to workflow run information
+      if: always() && steps.has_permissions.outputs.result == 'true'
+      uses: actions/github-script@v6
+      with:
+        github-token: ${{ steps.get_comment_bot_token.outputs.token }}
+        script: |
+          // There can be a delay between steps completing and this being reflected
+          // in information queried from REST API, therefore we poll for the list of
+          // jobs associated with workflow run at interval of 1 second until the
+          // status of the previous "Run test command(s)" step indicates completed or
+          // a maximum number of attempts have been reached
+          const maximum_attempts = 5;
+          let got_completed_step_info = false;
+          let attempts = 0;
+          let job, step;
+          while (!got_completed_step_info & attempts < maximum_attempts) {
+            await new Promise(r => setTimeout(r, 1000));  // Wait for 1 second
+            const { data: run_jobs } = await github.rest.actions.listJobsForWorkflowRun({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.runId,
+            });
+            const in_progress_jobs = run_jobs.jobs.filter(
+              job => job.status == "in_progress"
+            );  
+            // There should only be only one job in progress
+            if (in_progress_jobs.length > 1) {
+              throw "Multiple jobs triggered by same keyword";
+            }
+            job = in_progress_jobs[0];
+            const run_test_commands_steps = job.steps.filter(
+              step => step.name == "Run test command(s)"
+            );
+            // There should only be one step with name "Run test command(s)"
+            if (run_test_commands_steps.length > 1) {
+              throw "Multiple steps with name 'Run test command(s)'";
+            }
+            step = run_test_commands_steps[0];
+            got_completed_step_info = (step.status == "completed");
+            attempts += 1;
```

Comment:
I think the logic here will no longer work for the updated workflow as this specifically looks for a step with name `Run test command(s)` which will not exist for this workflow. Nearest equivalent is `Trigger profiling workflow` step but I'm not sure if the subsequent logic will follow through for this as its calling another reusable workflow rather than running a step directly (in particular the `step.status` check).

Reply:
I think this is fixed by our choice to recycle the `run-on-comment` workflow now